addons/pos_mercado_pago/models/pos_payment_method.py

```diff
@@ -0,0 +1,182 @@
+        # send a request to change the mode to Mercado Pago
+        mode = json.loads('{"operating_mode": "%s"}' % ("STANDALONE" if self.point_smart_op_mode == "standalone" else "PDV"))
+        # Whatever the Mercado Response...
+        self._call_mp("patch", f"/point/integration-api/devices/{self.mp_smart_point_ids.id_point_smart}", mode)
+        # We alert the user for consequence of the change
+        return {
+            'warning': {
+                'title': _("Warning"),
+                'message': _("Terminal in standalone mode will not work with the Odoo system") 
+                        if self.point_smart_op_mode == 'standalone' 
+                        else _("On the terminal, close and reopen the 'Caja' to apply the change"),
+            }
+        }           
+
+    """
+    Request stuff
+    """
+    def _call_mp(self, action: str, api_endpoint: str, json_data: json):
+        session = requests.Session()
+        session.mount('https://', requests.adapters.HTTPAdapter(max_retries=requests.adapters.Retry(
+            total=6,
+            backoff_factor=2,
+            status_forcelist=[202, 500, 502, 503, 504],
+        )))
+        session.headers.update({'Authorization': f"Bearer {self.mp_bearer_token}"})
+        try:
+            ep = 'https://api.mercadopago.com' + api_endpoint
+            resp = session.request(action, ep, json=json_data, timeout=TIMEOUT)
+            parsed = json.loads(resp.content)
+            return parsed
+        except requests.exceptions.RequestException as e:
+            return {'error': ("There are some issues between us and Mercado Pago, try again later.%s", e)}
+    """
+    """
+
+    def _is_write_forbidden(self, fields):
+        # Allow the modification of these fields even if a pos_session is open
+        whitelisted_fields = {'mp_last_message','mp_last_payment_intend_id'}
+        return bool(fields - whitelisted_fields and self.open_session_ids)
+
+    #  Called from frontend
+    def mp_payment_intend(self, infos):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
+        # Create a Mercado Pago payment intend for the current terminal
+        resp = self._call_mp("post", f"/point/integration-api/devices/{self.mp_smart_point_ids.id_point_smart}/payment-intents", infos)
+        _logger.info("data from mp: %s", resp)
+        # Store the payment intend id for further cancelation, if any.
+        # Here the response can be also "There is already a queued intent for the device"
+        if 'id' in resp:
+            self.mp_last_payment_intend_id = resp.get('id')
+        return resp
+
+    #  Called from frontend
+    def mp_get_last_message(self):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
```

Comment:
AccessError message is the good one or a copy/paste ?

Reply:
Pleas give me  some advise here since I just tried with simple user (demo) and the pos doesn't work

Reply:
You already need to be a pos user to access `pos.payment.method` objects. What do you want to check exactly
? Or you want to check that the user has write access (manager?)